### ISSUE TITLE
'import site' failed; use -v for traceback

### ISSUE THREAD

**Issue by adamjstewart** [LLNL/spack#878]
As the title suggests, I'm seeing a weird error message when I run any Spack command:
```
$ spack purge
'import site' failed; use -v for traceback
$ spack edit elk
'import site' failed; use -v for traceback
```
This usually happens after an install crashes, and it goes away as soon as I log out and log back in. Has anyone else seen this before?

**Comment by tgamblin** [comment]
@adamjstewart: is this still happening for you?  I have not seen this but wouldn't mind figuring out what's causing it.

**Comment by adamjstewart** [comment]
It only happens to me on rare occasions. So rare that reproducing it will likely be impossible. It's happened to me at least 2 or 3 times, but not recently.

I think the first time it happened was when I was working on the Elk package. When I tried building elk with `openblas~openmp`, `make test` prints out a warning message telling you that BLAS needs to be compiled with OpenMP support. But it keeps on printing this message. Forever. I didn't notice it at first because I wasn't building in verbose mode, and it eventually printed so many lines to `spack-build.out` that my /tmp filesystem ran out of space and everything crashed. After I deleted it, I got that error message every time I ran a Spack command.

Like I said, logging out and logging back in fixes this, and the bug seems rare enough that we shouldn't be too concerned. But I'm just as curious as you are as to what could be causing this. If I see it again, I'll dig further. Maybe I'll try grepping for `import site`.

**Comment by adamjstewart** [comment]
This happened to me again today. Was unable to recreate it though :(

**Comment by tgamblin** [comment]
@adamjstewart I looked at [this issue](http://stackoverflow.com/questions/10672731/python-import-site-failed) and it looks like that can happen when an unexpected uid/gid is encountered.  Does your spack install contain a directory owned by a nonexistent user/group?

**Comment by adamjstewart** [comment]
Hmm, it certainly shouldn't. Other Google results suggest something changing the PYTHONPATH or PYTHONHOME. I'll print out my env the next time this happens. It's so difficult to debug transient problems like this.

**Comment by adamjstewart** [comment]
Ah, I finally figured out how to reproduce the problem!

1. install a package
2. cd to the installation directory
3. uninstall the package
4. run any spack command

This usually occurred for me after a build crashed, I switched to the stage directory, took a look through some of the files, and ran `spack purge`.

Alternatively, this error message can be reproduced independent of Spack:

1. mkdir ~/foobar
2. cd ~/foobar
3. rmdir ~/foobar
4. `python -c 'print "hello world"'`

Since this problem has nothing to do with Spack, I'm closing this issue. Sorry for the confusion!
